app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -213,20 +215,55 @@ export default Component.extend(
       });
     },
 
+    _destroyTippyInstances() {
+      this.tippyInstances.forEach((instance) => {
+        instance.destroy();
+      });
+      this.tippyInstances = [];
+    },
+
     @on("didInsertElement")
     _composerEditorInit() {
       const $input = $(this.element.querySelector(".d-editor-input"));
 
       if (this.siteSettings.enable_mentions) {
         $input.autocomplete({
           template: findRawTemplate("user-selector-autocomplete"),
-          dataSource: (term) =>
-            userSearch({
+          dataSource: (term) => {
+            this._destroyTippyInstances();
+            return userSearch({
               term,
               topicId: this.topic?.id,
               categoryId: this.topic?.category_id || this.composer?.categoryId,
               includeGroups: true,
-            }),
+            }).then((result) => {
+              if (result?.users?.length > 0) {
+                result.users.forEach((user, index) => {
+                  if (user.status) {
+                    user.index = index;
+                    user.statusHtml = createUserStatusMessage(user.status, {
+                      showTooltip: true,
+                      showDescription: true,
+                    });
+                    this.tippyInstances.push(user.statusHtml._tippy);
```

Comment:
@jancernik @AndrewPrigorshnev this API doesn’t feel right `user.statusHtml._tippy`.

Reply:
It feels weird, but it's the way we have for accessing the instance. Otherwise, we cannot access it later to destroy it.
https://atomiks.github.io/tippyjs/v6/tippy-instance/#-property

Reply:
> this API doesn’t feel right `user.statusHtml._tippy`

Yeah, I do agree, the alternative would be to use an instance returned from calls to `tippy()`, like this:

    const instances1 = tippy('button');

But that would require a major refactoring of this solution. So we probably can leave it for now.